### DISCUSSION TITLE
Is it possible to pull the headings from the file

### DISCUSSION THREAD

**Discussion by JOSHUAJEBARAJ** [SilentVoid13/Templater#571] [Help]
Hello 👋
I am currently looking for a way to pull all the subheadings from the current file and put it under alias of the frontmatter
Instead of manually copying and pasting the all headings from the file , I want to autopopulate
I am looking something like 

```bash
alias : [tp.file.headings()]
```
Is it possible to do it via templater plugin 

![image](https://user-images.githubusercontent.com/19353065/155985463-25b98ffa-7e79-4081-aa85-4fda265b54f1.png)

**Comment by sntrenter** [comment]
You could probably pull the whole files text and then regex each line for a heading pattern or a specific heading pattern. or would that not work for what you're trying to do?

**Comment by AB1908** [comment]
```js
const outlineArray = app.metadataCache.getFileCache(await tp.file.find_tfile(tp.file.title)).headings;
```

**Reply by AB1908** [reply]
You may also want to look at the TOC plugin.

**Reply by JOSHUAJEBARAJ** [reply]
I gave a shot to the TOC plugin but I don' think it solve my problem because I want the headings as text under alias so I can search it

**Reply by AB1908** [reply]
I see. I recall there was some other quick switcher alternative that let you look for headers. Perhaps that may help as well.
